src/core/distributions/phylogenetics/tree/MultispeciesCoalescentUniformPrior.cpp

```diff
@@ -34,87 +36,186 @@ MultispeciesCoalescentUniformPrior::~MultispeciesCoalescentUniformPrior()
 
 MultispeciesCoalescentUniformPrior* MultispeciesCoalescentUniformPrior::clone( void ) const
 {
-    
+
     return new MultispeciesCoalescentUniformPrior( *this );
 }
 
 
 double MultispeciesCoalescentUniformPrior::computeLnCoalescentProbability(size_t k, const std::vector<double> &times, double begin_age, double end_age, size_t index, bool add_final_interval)
 {
-    
+
     if ( k == 1 ) return 0.0;
-    
+
     double theta_max = max_theta->getValue();
-    
+
     double current_time = begin_age;
-    double fn = 0.0;
+
     size_t n = times.size();
+    double nt = n;
+
+    // std::cout << "theta max: " << theta_max << std::endl;
+    // std::cout << "n: " << n << std::endl;
+
     for (size_t i=0; i<n; ++i)
     {
-        
         // now we do the computation
-        //a is the time between the previous and the current coalescences
+        // a is the time between the previous and the current coalescences
+
+        // std::cout << "current time: " << current_time << std::endl;
+
         double a = times[i] - current_time;
-        
+        current_time = times[i];
+
         // get the number j of individuals we had before the current coalescence
         size_t j = k - i;
-        double n_pairs = j * (j-1.0) / 2.0;
+        double n_pairs = j * (j-1.0);
 
         fn += a * n_pairs;
+        //
+        // std::cout << "j: " << j << std::endl;
+        // std::cout << "fn += " << a * n_pairs << std::endl;
     }
-    
+
     // compute the probability of no coalescent event in the final part of the branch
     // only do this if the branch is not the root branch
     if ( add_final_interval == true )
     {
         double final_interval = end_age - current_time;
         size_t j = k - times.size();
-        double n_pairs = j * (j-1.0) / 2.0;
```

Comment:
Why is the `/2.0` getting removed?

Reply:
This implementation does not work and I'm deferring to the correcting version in the `hsiang_dev` branch.